airbyte-config/config-models/src/main/java/io/airbyte/config/Configs.java

```diff
@@ -541,13 +559,73 @@ public interface Configs {
    */
   boolean shouldRunConnectionManagerWorkflows();
 
+  /**
+   * Define if the worker is operating as a Control Plane worker, a Data Plane worker, or as a
+   * combined worker. - Control plane workers process tasks related to control-flow, like scheduling
+   * and routing. - Data plane workers process tasks related to data syncing, and are typically
+   * deployed outside of Airbyte's internal network. - Combined workers process all tasks, and are the
+   * default for OSS deployments. - Internal-use only.
+   */
+  WorkerPlane getWorkerPlane();
+
+  // Worker - Control Plane configs
+
+  /**
+   * Return true if the WorkerPlane is CONTROL_PLANE or COMBINED. Derived from the result of
+   * {@link Configs#getWorkerPlane()}. Internal-use only.
+   */
+  boolean isControlPlaneWorker();
+
+  /**
+   * TEMPORARY: Define a set of connection IDs that should run in Airbyte's AWS Data Plane. - This
+   * should only be set on Control-plane workers, since those workers decide which Data Plane task
+   * queue to use based on connectionId. - Will be removed in favor of the Routing Service in the
+   * future. Internal-use only.
+   */
+  Set<String> connectionIdsForDataPlane();
+
+  // Worker - Data Plane configs
+
+  /**
+   * Return true if the WorkerPlane is DATA_PLANE or COMBINED. Derived from the result of
+   * {@link Configs#getWorkerPlane()}. Internal-use only.
+   */
+  boolean isDataPlaneWorker();
```

Comment:
@cgardens discussion from parent PR:
![image](https://user-images.githubusercontent.com/7798112/186285543-182015fe-3be9-497f-8ad3-eef6d4c0695f.png)

Reply:
1. Yeah these helpers are probably superfluous, I can remove them
2. For me a `worker` is a container running an `airbyte-worker` image, so a `ControlPlaneWorker` is an `airbyte-worker` instance that resides within Airbyte's Control Plane. Does that make sense? Basically, `ControlPlane` refers to a location here. Does that make sense with your mental model?

Reply:
okay. I'm convinced that current naming is okay.